File: services/orchestrator/frontend/src/types/agentMetrics.ts
export interface AgentMetrics {
  agentId: string
  agentName: string
  taskCount: number
  successCount: number
  failedCount: number
  retryCount: number
  averageDuration: number
  lastExecutedAt: string | null
  repairCount: number
  changeCount: number
}

export interface AgentExecutionMetric {
  taskId: string
  agentId: string
  executionId: string
  durationMillis: number
  status: string
  createdAt: string | null
}

export interface AgentMetricsDetail {
  metrics: AgentMetrics
  executions: AgentExecutionMetric[]
}

export interface TaskExecutionMetrics {
  taskId: string
  taskStatus: string
  executionCount: number
  successCount: number
  failedCount: number
  totalDurationMillis: number
  averageDurationMillis: number
  repairCount: number
  retryCount: number
  changeCount: number
  approvedChanges: number
  rejectedChanges: number
  reviewPassRate: number
  executions: AgentExecutionMetric[]
}

File: services/orchestrator/src/main/java/com/aidevos/orchestrator/metrics/agent/AgentMetrics.java
package com.aidevos.orchestrator.metrics.agent;

import java.time.Instant;

/**
 * Aggregated execution metrics for one agent, derived on demand from the
 * existing ExecutionRecords, audit events, repair tasks and change sets. Never
 * stored as business data.
 */
public record AgentMetrics(
		String agentId,
		String agentName,
		int taskCount,
		int successCount,
		int failedCount,
		int retryCount,
		long averageDuration,
		Instant lastExecutedAt,
		int repairCount,
		int changeCount,
		long tokenCount,
		double estimatedCost,
		double successCost) {

	/**
	 * Backward-compatible constructor without the usage figures (token
	 * consumption, estimated cost and success cost default to zero).
	 */
	public AgentMetrics(String agentId, String agentName, int taskCount, int successCount,
			int failedCount, int retryCount, long averageDuration, Instant lastExecutedAt,
			int repairCount, int changeCount) {
		this(agentId, agentName, taskCount, successCount, failedCount, retryCount,
			averageDuration, lastExecutedAt, repairCount, changeCount, 0, 0.0, 0.0);
	}
}
